src/sqlfluff/rules/ambiguous/AM08.py

```diff
@@ -0,0 +1,95 @@
+"""Implementation of Rule AM08."""
+
+from typing import Optional, Tuple
+
+from sqlfluff.core.parser import (
+    ComparisonOperatorSegment,
+    KeywordSegment,
+    LiteralSegment,
+    WhitespaceSegment,
+)
+from sqlfluff.core.rules import BaseRule, LintFix, LintResult, RuleContext
+from sqlfluff.core.rules.crawlers import SegmentSeekerCrawler
+
+
+class Rule_AM08(BaseRule):
+    """Join clauses should be present.
+
+    **Anti-pattern**
+
+    A join is used without specifying join condition.
+    This can cause unintended cross join behavior.
+    Prefer explicit const condition instead.
+
+    .. code-block:: sql
+       :force:
+
+        SELECT
+            foo
+        FROM bar
+        JOIN baz;
+
+    **Best practice**
+
+    Use const condition instead.
+
+    .. code-block:: sql
+       :force:
+
+        SELECT
+            foo
+        FROM bar
+        JOIN baz
+        ON 1=1;
+
+    If dialect allows, alternatively use CROSS JOIN.
+
+    .. code-block:: sql
+       :force:
+
+        SELECT
+            foo
+        FROM bar
+        CROSS JOIN baz;
```

Comment:
I like this. If someone _wants_ a `cross join` they should explicitly ask for it.